src/ui/TextComponent.cpp

```diff
@@ -0,0 +1,69 @@
+#include "TextComponent.h"
+
+TextComponent::~TextComponent() { }
+
+void TextComponent::setOwner(VisualElement* owner) {
+    _owner = owner;
+}
+
+const String& TextComponent::getDisplayString() {
+    return _displayString;
+}
+
+TextComponent& TextComponent::setDisplayString(const String& str) {
+    _displayString = str;
+    _textChanged = true;
+    return *this;
+}
+
+TextComponent& TextComponent::setFont(const GFXfont* font) {
+    _font = font;
+    _textChanged = true;
+    return *this;
+}
+
+TextComponent& TextComponent::setFontColour(const uint16_t colour) {
+    _fontColour = colour;
+    return *this;
+}
+
+uint16_t TextComponent::getFontColour() {
+    return _fontColour;
+}
+
+TextComponent& TextComponent::setFontSize(uint8_t w, uint8_t h) {
+    _textSizeX = w;
+    _textSizeY = h;
+    _textChanged = true;
+    return *this;
+}
+
+void TextComponent::draw(Adafruit_GFX& display) {
+    if (_textChanged) {
+        Point dimensions = ui_util::computeStringDimensions(_font, _displayString, _textSizeX, _textSizeY);
+        _textWidth = dimensions.x;
+        _textHeight = dimensions.y;
+        _textChanged = false;
+    }
+
+    _drawInternal(display, _fontColour);
+}
+
+void TextComponent::clear(Adafruit_GFX& display) {
+    _drawInternal(display, _owner->getBackgroundColour());
```

Comment:
```VisualElement::clear``` will most likely only be used for transitioning between frames in animations (picture a blinking animation that indicates that the BPM is on).

Good catch though.  ```TextComponent::clear``` should be also be called by TextElement and UIButton and it is not

Reply:
After looking at this again, it makes sense to me that it isn't called anywhere just yet because there aren't any examples of text animations.  I've added a colour parameter to ```TextComponent::clear``` for controllability, but this is definitely one of those things that might have to change depending on how it's used